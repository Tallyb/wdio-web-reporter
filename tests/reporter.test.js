import WebReporter from '../src/reporter';

import { SUITE_FEATURE, SUITE_SCENARIO, TEST_PASS, TEST_FAIL } from './fixtures';

jest.mock('sync-request');
import request from 'sync-request';

describe('webhook-reporter', () => {
    const baseReporter = {};
    const options = {url: 'dummy'};
    let config = {specs: {
        a: false,
        b: 1
    }};

    beforeEach(() => {
        request.mockReset();
    });

    it('should initialize the reporter', () => {        
        let reporter = new WebReporter(baseReporter, config, options);
        expect(reporter.options).toEqual(options);
        expect(reporter.config).toEqual(config);
    });

    it('should send results if url provided', () => {
        let reporter = new WebReporter(baseReporter, config, {url: 'dummy'});
        reporter.emit('end');
        expect(request).toHaveBeenCalled();

    });

    it('should not send results if no url provided', () => {
        let reporter = new WebReporter(baseReporter, config, {});
        reporter.emit('end');
        expect(request).not.toHaveBeenCalled();
    });

    it('should send environment variables', () => {
        process.env.DUMMY='MY_TEST_DUMMY';
        let reporter = new WebReporter(baseReporter, config, {
            url: 'https://dummy.com',
            environment: ['DUMMY']
        });
        reporter.emit('end');
        expect(request).toHaveBeenCalled();
        expect(request.mock.calls[0]).toMatchSnapshot();
        expect(request.mock.calls[0][2].json.environment).toEqual({DUMMY: 'MY_TEST_DUMMY'});
    });

    it('should emit results', () => {
        let reporter = new WebReporter(baseReporter, config, options);
        reporter.emit('runner:start');
        reporter.emit('suite:start', SUITE_FEATURE);
        reporter.emit('suite:start', SUITE_SCENARIO);
        reporter.emit('test:pass', TEST_PASS);
        reporter.emit('test:pass', TEST_PASS);
        reporter.emit('test:fail', TEST_FAIL);
        reporter.emit('test:pending', TEST_PASS);
        reporter.emit('runner:end');
        reporter.emit('end');
        expect(request).toHaveBeenCalled();
        expect(request.mock.calls[0]).toMatchSnapshot();
    });

});
