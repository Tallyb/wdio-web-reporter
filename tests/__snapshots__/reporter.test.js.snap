// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webhook-reporter should emit results 1`] = `
Array [
  "POST",
  "dummy",
  Object {
    "json": Object {
      "environment": Object {},
      "results": Object {
        "a6c6e5d30799348316f0995d2b867b7b": Object {
          "failed": 1,
          "failures": Array [
            Object {
              "error": "it failed",
              "title": "Following locations exist: \\"Chandelier, La Fayette\\"",
            },
          ],
          "passed": 2,
          "skipped": 1,
          "title": "Test feature",
        },
      },
    },
  },
]
`;

exports[`webhook-reporter should send environment variables 1`] = `
Array [
  "POST",
  "https://dummy.com",
  Object {
    "json": Object {
      "environment": Object {
        "DUMMY": "MY_TEST_DUMMY",
      },
      "results": Object {},
    },
  },
]
`;
